### PR TITLE
Mohamed test

### PR DESCRIPTION
**Pull request by mohamedAlmukhtar**
- Added Recurring Events feature
- Added dropdown for rsvp display (Event#show, maincalendar)
- Only one rsvp per event allowed
- Email rsvps show on rsvp display
- 4 Validation and 2 Association tests (merchandise_test.rb)
- One Integeration test for email rsvps (user_visits_event.rb)
- Two model tests for Recurring Events (events_test.rb)

### CODE REVIEW

**Comment by RogelioMart** [comment]
TL/DR
the new feature is great and has no issue but many other things are broken. I do not recommend merging this branch to master until the previously described errors are fixed.

REPORT
Overall the new feature works like a charm and does not affect other events that are not recurring.
But there are some issues.
I cannot access my view profile. 
When I try to save a buy/donate, try to change anything on my profile such as adding a picture, changing topics or my email and password I get an error. (To see the error for all of these check the picture below.) Oddly enough though the error does not seem to occur if there are no event queued up.
Another odd thing is that despite that I get an error every time I try to change anything on my profile or on rewards Those things still get saved and even send me an email if I did something like changing my email.

mError


But the recurring event feature works like a charm. There are no errors with it and seems like it will be very helpful for the user when creating multiple events.

■test/integration/events/user_visits_event.rb  all had errors with the following message.
	ActionView::Template::Error: The @users variable appears to be empty. Did you forget to pass 	the collection object for will_paginate?
    		app/views/layouts/_maincalendar.html.erb:4
    		app/views/users/show.html.erb:44

■test/integration/users/home_page/guest_visits_home_page_test.rb was unable to run displayed the following errors. 
	/home/user/thinQStuff/mTest/tomatocan/test/integration/users/home_page/	guest_visits_home_page_test.rb:103: syntax error, unexpected ===, expecting `end'
	=======
	^~~
	/home/user/thinQStuff/mTest/tomatocan/test/integration/users/home_page/	guest_visits_home_page_test.rb:107: syntax error, unexpected `end', expecting end-of-input
	  end 
	  ^~~

■test/models/events_test.rb had 15 assertions 1 failure
	EventTest#test_endat_greaterthan_startat 	[/home/user/thinQStuff/mTest/tomatocan/test/models/events_test.rb:97]:
	Accepting events with start_at > end_at
	
	I recommend deleting this test because the user does not set the end_at date anymore.

■test/models/merchandise_test.rb 28 assertions 1 failure
	MerchandiseTest#test_error_message_for_name_presence 	[/home/user/thinQStuff/mTest/tomatocan/test/models/merchandise_test.rb:33]:
	no validation for presence of name.
	Expected [] to be nil

■test/models/rsvpq_test.rb: 4 assertions 0 failures

■test/models/user_test.rb 88 assertions 3 errors

	Error:
	TestUser#test_get_youtube_id_test:
	NoMethodError: undefined method `get_youtube_id' for #<User:0x0000555ceff98160>
	    test/models/user_test.rb:138:in `block in <class:TestUser>'


	rails test test/models/user_test.rb:132

	..E

	Error:
	TestUser#test_videodesc_must_be_less_than255char:
	ActiveModel::UnknownAttributeError: unknown attribute 'videodesc1' for User.
	    test/models/user_test.rb:102:in `block (2 levels) in <class:TestUser>'
	    test/models/user_test.rb:101:in `each'
	    test/models/user_test.rb:101:in `block in <class:TestUser>'


	rails test test/models/user_test.rb:100

	......E

	Error:
	TestUser#test_parse_youtube:
	NoMethodError: undefined method `get_youtube_id' for #<User:0x0000555cf0308b38>
	    test/models/user_test.rb:128:in `block in <class:TestUser>'


	rails test test/models/user_test.rb:124



To Summarize, the new feature is great and has no issue but many other things are broken. I do not recommend merging this branch to master until the previously described errors are fixed.

**Comment by kmyee97** [comment]
Great features to have on the site. However, there are some minor errors.

[test_result.docx](https://github.com/ece106/tomatocan/files/4915624/test_result.docx)

TL/DR: Some of the tests are not working. Also, there are some minor design issues.

**Comment by kmyee97** [comment]
Furthermore, rsvping to a recurring event will automatically rsvp to that event x N.
